### ISSUE TITLE
[ghcr.io/napari/napari-xpra] No napari screen shown, just empty window, also napari 0.4.17 (macOS 13.4, Safari 16.5.2)

### ISSUE THREAD

**Issue by psobolewskiPhD** [napari/napari#6094]
## 🐛 Bug

Edit: **this only appears to affect Safari and may be solved in the future. In the meantime, Chrome works!** 
See: https://github.com/napari/napari/issues/6094#issuecomment-1653877731

The `napari-xpra` image runs but results in an empty window named `napari` and errors:
<img width="1419" alt="image" src="https://github.com/napari/napari/assets/76622105/89d328d0-4821-4796-9676-361264358f85">

The most relevant errors seem to be:
```
2023-07-25 15:46:11 2023-07-25 19:46:11,508 xpra is ready.
2023-07-25 15:46:11 2023-07-25 19:46:11,903 Warning: ffmpeg encoder testing failed with h264+mp4:
2023-07-25 15:46:11 2023-07-25 19:46:11,903  Invalid argument
2023-07-25 15:46:11 2023-07-25 19:46:11,916 Warning: ffmpeg encoder testing failed with mpeg4+mp4:
2023-07-25 15:46:11 2023-07-25 19:46:11,918  Invalid argument
2023-07-25 15:46:11 2023-07-25 19:46:11,921 Warning: ffmpeg encoder testing failed with mpeg1:
2023-07-25 15:46:11 2023-07-25 19:46:11,921  Invalid argument
2023-07-25 15:46:11 2023-07-25 19:46:11,924 Warning: ffmpeg encoder testing failed with mpeg2:
2023-07-25 15:46:11 2023-07-25 19:46:11,924  Invalid argument
2023-07-25 15:46:11 2023-07-25 19:46:11,924 ffmpeg: all the codecs have failed! (h264+mp4, mpeg4+mp4, mpeg1, mpeg2)
```
and

```
2023-07-25 15:47:28 2023-07-25 19:47:28,025 Error during encoding:
2023-07-25 15:47:28 Traceback (most recent call last):
2023-07-25 15:47:28   File "/usr/lib/python3/dist-packages/xpra/server/source/client_connection.py", line 300, in encode_loop
2023-07-25 15:47:28     fn_and_args[1](*fn_and_args[2:])
2023-07-25 15:47:28   File "/usr/lib/python3/dist-packages/xpra/server/window/windowicon_source.py", line 225, in compress_and_send_window_icon
2023-07-25 15:47:28     image = image.resize((rw, rh), Image.ANTIALIAS)
2023-07-25 15:47:28 AttributeError: module 'PIL.Image' has no attribute 'ANTIALIAS'
2023-07-25 15:47:28 2023-07-25 19:47:28,034 Error during encoding:
2023-07-25 15:47:28 Traceback (most recent call last):
2023-07-25 15:47:28   File "/usr/lib/python3/dist-packages/xpra/server/source/client_connection.py", line 300, in encode_loop
2023-07-25 15:47:28     fn_and_args[1](*fn_and_args[2:])
2023-07-25 15:47:28   File "/usr/lib/python3/dist-packages/xpra/server/window/windowicon_source.py", line 225, in compress_and_send_window_icon
2023-07-25 15:47:28     image = image.resize((rw, rh), Image.ANTIALIAS)
2023-07-25 15:47:28 AttributeError: module 'PIL.Image' has no attribute 'ANTIALIAS'
```
This 2nd one has been fixed at Xpra, but not yet released: https://github.com/Xpra-org/xpra/commit/4f8a1b195d798416b5eedabe57402015552e45f2

Also, strangely, the napari:0.4.18 container has napari 0.4.17 if I check via the Terminal in docker desktop:
```
# pip show napari
Name: napari
Version: 0.4.17
Summary: n-dimensional array viewer in Python
Home-page: https://napari.org
Author: napari team
Author-email: napari-steering-council@googlegroups.com
License: BSD 3-Clause
Location: /usr/local/lib/python3.10/dist-packages
Requires: app-model, appdirs, cachey, certifi, dask, imageio, jsonschema, magicgui, napari-console, napari-plugin-engine, napari-svg, npe2, numpy, numpydoc, pandas, Pillow, pint, psutil, psygnal, pydantic, pygments, PyOpenGL, PyYAML, qtpy, scikit-image, scipy, sphinx, superqt, tifffile, toolz, tqdm, typing-extensions, vispy, wrapt
Required-by:
```

Full log:
<details>

```
2023-07-25 15:46:07 Launching napari on Xpra. Connect via http://localhost:9876
2023-07-25 15:46:07 
2023-07-25 15:46:07 Warning: running as root
2023-07-25 15:46:08 
2023-07-25 15:46:08 Warning: running as root
2023-07-25 15:46:08 failed to initialize Gtk, no display?
2023-07-25 15:46:08 2023-07-25 19:46:08,379 created tcp socket '0.0.0.0:9876'
2023-07-25 15:46:08 2023-07-25 19:46:08,386 no uinput module (not usually needed)
2023-07-25 15:46:10 2023-07-25 19:46:10,594 pointer device emulation using XTest
2023-07-25 15:46:10 2023-07-25 19:46:10,733 serving html content from '/usr/share/xpra/www'
2023-07-25 15:46:10 2023-07-25 19:46:10,733 wrote pid 7 to '/tmp/100/server.pid'
2023-07-25 15:46:10 2023-07-25 19:46:10,771 Warning: socket directory '/tmp/100'
2023-07-25 15:46:10 2023-07-25 19:46:10,771  expected permissions 700 but found 750
2023-07-25 15:46:10 2023-07-25 19:46:10,776 created unix domain socket '/run/user/0/xpra/ab85beb896a3-100'
2023-07-25 15:46:10 2023-07-25 19:46:10,776 created unix domain socket '/run/xpra/ab85beb896a3-100'
2023-07-25 15:46:10 2023-07-25 19:46:10,777 created unix domain socket '/root/.xpra/ab85beb896a3-100'
2023-07-25 15:46:10 2023-07-25 19:46:10,777 created unix domain socket '/tmp/100/socket'
2023-07-25 15:46:10 2023-07-25 19:46:10,777 xvfb pid=13
2023-07-25 15:46:10 2023-07-25 19:46:10,832 Warning: cannot set resolution to (8192, 4096)
2023-07-25 15:46:10 2023-07-25 19:46:10,833  (this resolution is not available)
2023-07-25 15:46:11 
2023-07-25 15:46:11 Warning: running as root
2023-07-25 15:46:11 2023-07-25 19:46:11,326 watching for applications menu changes in:
2023-07-25 15:46:11 2023-07-25 19:46:11,326  '/usr/share/applications'
2023-07-25 15:46:11 2023-07-25 19:46:11,342 Warning: webcam forwarding is disabled
2023-07-25 15:46:11 2023-07-25 19:46:11,343  the virtual video directory '/sys/devices/virtual/video4linux' was not found
2023-07-25 15:46:11 2023-07-25 19:46:11,343  make sure that the 'v4l2loopback' kernel module is installed and loaded
2023-07-25 15:46:11 2023-07-25 19:46:11,351  or use the 'webcam=no' option
2023-07-25 15:46:11 2023-07-25 19:46:11,351 found 0 virtual video devices for webcam forwarding
2023-07-25 15:46:11 2023-07-25 19:46:11,411 Error parsing xdg menu data:
2023-07-25 15:46:11 2023-07-25 19:46:11,413  ParsingError in file '/etc/xdg/menus/debian-menu.menu', File not found
2023-07-25 15:46:11 2023-07-25 19:46:11,413  this is either a bug in python-xdg,
2023-07-25 15:46:11 2023-07-25 19:46:11,414  or an invalid system menu configuration
2023-07-25 15:46:11 2023-07-25 19:46:11,438 Warning: lpinfo command failed and returned 1
2023-07-25 15:46:11 2023-07-25 19:46:11,438  command used: '/usr/sbin/lpinfo --make-and-model Generic PDF Printer -m'
2023-07-25 15:46:11 2023-07-25 19:46:11,496 Warning: lpinfo command failed and returned 1
2023-07-25 15:46:11 2023-07-25 19:46:11,497  command used: '/usr/sbin/lpinfo --make-and-model Generic PostScript Printer -m'
2023-07-25 15:46:11 2023-07-25 19:46:11,499 printer forwarding enabled using pdf
2023-07-25 15:46:11 2023-07-25 19:46:11,506 waiting for initialization thread to complete
2023-07-25 15:46:11 2023-07-25 19:46:11,507 started command `python3 -m napari` with pid 58
2023-07-25 15:46:11 2023-07-25 19:46:11,508 xpra is ready.
2023-07-25 15:46:11 2023-07-25 19:46:11,903 Warning: ffmpeg encoder testing failed with h264+mp4:
2023-07-25 15:46:11 2023-07-25 19:46:11,903  Invalid argument
2023-07-25 15:46:11 2023-07-25 19:46:11,916 Warning: ffmpeg encoder testing failed with mpeg4+mp4:
2023-07-25 15:46:11 2023-07-25 19:46:11,918  Invalid argument
2023-07-25 15:46:11 2023-07-25 19:46:11,921 Warning: ffmpeg encoder testing failed with mpeg1:
2023-07-25 15:46:11 2023-07-25 19:46:11,921  Invalid argument
2023-07-25 15:46:11 2023-07-25 19:46:11,924 Warning: ffmpeg encoder testing failed with mpeg2:
2023-07-25 15:46:11 2023-07-25 19:46:11,924  Invalid argument
2023-07-25 15:46:11 2023-07-25 19:46:11,924 ffmpeg: all the codecs have failed! (h264+mp4, mpeg4+mp4, mpeg1, mpeg2)
2023-07-25 15:46:11 2023-07-25 19:46:11,985 7.7GB of system memory
2023-07-25 15:46:12 2023-07-25 19:46:12,002 GStreamer version 1.20.3
2023-07-25 15:46:14 2023-07-25 19:46:14,632 OpenGL is supported on display ':100'
2023-07-25 15:46:14 2023-07-25 19:46:14,633  using 'llvmpipe (LLVM 15.0.7, 128 bits)' renderer
2023-07-25 15:46:14 2023-07-25 19:46:14,641 started command `ibus-daemon` with pid 89
2023-07-25 15:46:16 WARNING: QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
2023-07-25 15:46:16 19:46:16 WARNING QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
2023-07-25 15:46:16 WARNING: QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
2023-07-25 15:46:16 19:46:16 WARNING QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
2023-07-25 15:46:16 2023-07-25 19:46:16,875 Error parsing xdg menu data:
2023-07-25 15:46:16 2023-07-25 19:46:16,875  ParsingError in file '/etc/xdg/menus/debian-menu.menu', File not found
2023-07-25 15:46:16 2023-07-25 19:46:16,875  this is either a bug in python-xdg,
2023-07-25 15:46:16 2023-07-25 19:46:16,875  or an invalid system menu configuration
2023-07-25 15:46:19 2023-07-25 19:46:19,277 xpra X11 seamless version 4.4.6-r29 64-bit
2023-07-25 15:46:19 2023-07-25 19:46:19,278  uid=0 (root), gid=0 (root)
2023-07-25 15:46:19 2023-07-25 19:46:19,278  running with pid 7 on Linux Ubuntu 22.04 jammy
2023-07-25 15:46:19 2023-07-25 19:46:19,280  connected to X11 display :100 with 24 bit colors
2023-07-25 15:47:27 2023-07-25 19:47:27,051 Handshake complete; enabling connection
2023-07-25 15:47:27 2023-07-25 19:47:27,115 HTML5 Mac OSX Safari client version 8.1
2023-07-25 15:47:27 2023-07-25 19:47:27,127 setting keyboard layout to 'us'
2023-07-25 15:47:27 2023-07-25 19:47:27,346  client root window size is 1307x1057
2023-07-25 15:47:27 2023-07-25 19:47:27,346   HTML (346x280 mm - DPI: 96x96)
2023-07-25 15:47:27 2023-07-25 19:47:27,346     Canvas
2023-07-25 15:47:27 2023-07-25 19:47:27,894 server virtual display now set to 1307x1057
2023-07-25 15:47:27 2023-07-25 19:47:27,911  automatic picture encoding enabled, also available:
2023-07-25 15:47:27 2023-07-25 19:47:27,911   jpeg, png, png/P, png/L, rgb32, rgb24, webp, avif
2023-07-25 15:47:27 2023-07-25 19:47:27,920 Error parsing xdg menu data:
2023-07-25 15:47:27 2023-07-25 19:47:27,921  ParsingError in file '/etc/xdg/menus/debian-menu.menu', File not found
2023-07-25 15:47:27 2023-07-25 19:47:27,921  this is either a bug in python-xdg,
2023-07-25 15:47:27 2023-07-25 19:47:27,921  or an invalid system menu configuration
2023-07-25 15:47:27 2023-07-25 19:47:27,959 Warning: failed to add virtual printer 'HTML5 client'
2023-07-25 15:47:27 2023-07-25 19:47:27,960  failed to connect to server
2023-07-25 15:47:27 2023-07-25 19:47:27,969 DPI set to 96 x 96
2023-07-25 15:47:28 2023-07-25 19:47:28,025 Error during encoding:
2023-07-25 15:47:28 Traceback (most recent call last):
2023-07-25 15:47:28   File "/usr/lib/python3/dist-packages/xpra/server/source/client_connection.py", line 300, in encode_loop
2023-07-25 15:47:28     fn_and_args[1](*fn_and_args[2:])
2023-07-25 15:47:28   File "/usr/lib/python3/dist-packages/xpra/server/window/windowicon_source.py", line 225, in compress_and_send_window_icon
2023-07-25 15:47:28     image = image.resize((rw, rh), Image.ANTIALIAS)
2023-07-25 15:47:28 AttributeError: module 'PIL.Image' has no attribute 'ANTIALIAS'
2023-07-25 15:47:28 2023-07-25 19:47:28,034 Error during encoding:
2023-07-25 15:47:28 Traceback (most recent call last):
2023-07-25 15:47:28   File "/usr/lib/python3/dist-packages/xpra/server/source/client_connection.py", line 300, in encode_loop
2023-07-25 15:47:28     fn_and_args[1](*fn_and_args[2:])
2023-07-25 15:47:28   File "/usr/lib/python3/dist-packages/xpra/server/window/windowicon_source.py", line 225, in compress_and_send_window_icon
2023-07-25 15:47:28     image = image.resize((rw, rh), Image.ANTIALIAS)
2023-07-25 15:47:28 AttributeError: module 'PIL.Image' has no attribute 'ANTIALIAS'
2023-07-25 15:47:28 2023-07-25 19:47:28,044 client   1 received xdg start menu data
2023-07-25 15:47:28 2023-07-25 19:47:28,049 client   1 [object Object] 1 780 360 360 360 [object Object] false false [object Object] window=>this.send_configure_window(window,{},!1) (event,window)=>this.on_mousemove(event,window) (event,window)=>this.on_mousedown(event,window) (event,window)=>this.on_mouseup(event,window) (event,window)=>this.on_mousescroll(event,window) window=>this.set_focus(window) window=>this.send_close_window(window) 1
2023-07-25 15:47:28 2023-07-25 19:47:28,062 client   1 [object Object] 2 10 96 1086 656 [object Object] false false [object Object] window=>this.send_configure_window(window,{},!1) (event,window)=>this.on_mousemove(event,window) (event,window)=>this.on_mousedown(event,window) (event,window)=>this.on_mouseup(event,window) (event,window)=>this.on_mousescroll(event,window) window=>this.set_focus(window) window=>this.send_close_window(window) 1
2023-07-25 15:47:28 2023-07-25 19:47:28,063 client   1 startup complete
2023-07-25 15:47:28 2023-07-25 19:47:28,063 client   1 connection-established
2023-07-25 15:47:28 2023-07-25 19:47:28,063 client   1 update encodings: encoding.rgb.version,encoding.pillow.version,encoding.webp.version,encoding.jpeg.version,encoding.avif.version,encoding.x264.version,encoding.vpx.version,encoding.ffmpeg.version,encoding.swscale.version,encoding.cython.version,encodings,encodings.core,encodings.allowed,encodings.lossless,encodings.with_speed,encodings.with_quality,encodings.with_lossless_mode
2023-07-25 15:47:28 2023-07-25 19:47:28,063 client   1 audio media source open
2023-07-25 15:47:28 2023-07-25 19:47:28,063 client   1 using audio codec string for mp3: audio/mpeg
2023-07-25 15:47:28 2023-07-25 19:47:28,063 client   1 audio: requesting mp3 stream from the server
2023-07-25 15:47:28 2023-07-25 19:47:28,087 no source plugins available
2023-07-25 15:47:28 2023-07-25 19:47:28,087 failed to setup 'auto' sound stream source
2023-07-25 15:47:28 2023-07-25 19:47:28,092 client   1 received end-of-stream from server
2023-07-25 15:47:28 2023-07-25 19:47:28,092 client   1 audio: stopping stream
2023-07-25 15:47:28 2023-07-25 19:47:28,093 client   1 close_audio_mediasource: audio_source_buffer=[object SourceBuffer], media_source=[object MediaSource], audio=[object HTMLAudioElement]
2023-07-25 15:47:28 2023-07-25 19:47:28,094 client   1 audio-state: stopped
2023-07-25 15:47:32 2023-07-25 19:47:32,714 client   1 received xdg start menu data

```
</details>

## To Reproduce

1. `docker run -it --rm -p 9876:9876 ghcr.io/napari/napari-xpra:0.4.18`

## Expected behavior

The napari window should be functional in the browser, as described in the docs:
https://napari.org/stable/howtos/docker.html

## Environment

N/A

## Additional context

CC: @jaimergp

**Comment by jaimergp** [comment]
Oh, I am using Chrome and you are using Safari. I can reproduce if I open it with Safari, and it seems to be related to this bug: https://github.com/Xpra-org/xpra-html5/issues/227. So... nothing we can do til Safari is fixed ([the fix apparently went in in webkit](https://github.com/WebKit/WebKit/pull/11534), but it's not yet in Safari).

**Comment by psobolewskiPhD** [comment]
> Oh, I am using Chrome and you are using Safari. I can reproduce if I open it with Safari, and it seems to be related to this bug: [Xpra-org/xpra-html5#227](https://github.com/Xpra-org/xpra-html5/issues/227). So... nothing we can do til Safari is fixed ([the fix apparently went in in webkit](https://github.com/WebKit/WebKit/pull/11534), but it's not yet in Safari).

Awesome! I will edit the OP to clarify and ask a colleague to check on Windows and we should be good to go here!

**Comment by psobolewskiPhD** [comment]
@jaimergp shall we make a new issue for the version thing? just to keep track of it with this one closed?

**Comment by jaimergp** [comment]
I am opening a PR now. If you open a new issue, I'll point to that one; if not, this one is good enough for me.
